src/sparse/KokkosSparse_sptrsv_supernode.hpp

```diff
@@ -51,7 +51,9 @@
 #ifndef KOKKOSSPARSE_SPTRSV_SUPERNODE_HPP_
 #define KOKKOSSPARSE_SPTRSV_SUPERNODE_HPP_
 
-#if defined(KOKKOSKERNELS_ENABLE_TPL_LAPACKE) && defined(KOKKOSKERNELS_ENABLE_TPL_CBLAS)
+#if (defined(KOKKOSKERNELS_ENABLE_TPL_LAPACKE) && \
+    defined(KOKKOSKERNELS_ENABLE_TPL_CBLAS)) \
+    || defined(KOKKOSKERNELS_ENABLE_TPL_SUPERLU)
```

Comment:
Is this right to include cblas and lapacke when SuperLU is defined ?

Reply:
I do not think it is the correct place, I think these includes should be hidden within `KokkosBlas` and I intend to do this while resolving #589. These small changes are just to fix compilation errors. `|| defined(KOKKOSKERNELS_ENABLE_TPL_SUPERLU)` makes the definition of `read_supernodal_graphL` available for `sptrsv_superlu`.

Reply:
Thanks!

Reply:
Thank you for looking at this. By the way, this file uses CBLAS/LAPACKE, but not SuperLU (in fact, it is also used with sptrsv_cholmod). Does CBLAS/LAPACKE gets enabled, if SuperLU get enabled?

Reply:
No problem :) 

The problem here is that sptrsv_superlu needs the definition of `read_supernodal_graphL` -- otherwise, kokkos-kernels will not compile.

Reply:
So, you are trying to build it without CBLAS/LAPACKE enabled?

Reply:
Thanks for the PR feedback. This is my mistake. It looks like I had a bug with my build makefile generation scripts where CBLAS and LAPACKE were not enabled but SUPERLU was.

I now see that the perf tests pull in `KokkosKernels_config.h` via `KokkosSparse_sptrsv.hpp->KokkosSparse_sptrsv_symbolic_spec.hpp`.

Reply:
> So, you are trying to build it without CBLAS/LAPACKE enabled?

@iyamazaki: Yes, I was. This compilation issue arises when `--with-tpls=cholmod,superlu`.

Reply:
I am confused now. What is the goal here ? If you enable superlu, cholmod then cblas/lapacke are required. Your changes will get rid of cblas/lapacke dependency, but we are not there yet. Correct ?

Reply:
Yea, I don't think we can compile the code without enabling CBLAS/LAPACKE, right now..

Reply:
@srajama1: I was confused as well -- if we use `--with-tpls=cholmod,superlu`, one way to get things compiling is by adding `|| defined(KOKKOSKERNELS_ENABLE_TPL_SUPERLU)` here to fix a seemingly unrelated compilation error. If we use `--with-tpls=cholmod,superlu,cblas,blas,lapacke` everything works as expected without adding `|| defined(KOKKOSKERNELS_ENABLE_TPL_SUPERLU)` here.

> What is the goal here ?

The goal of this PR is to learn the kokkos-kernels PR guidelines and fix the build for the superlu and cholmod perf_tests and in the process.

> If you enable superlu, cholmod then cblas/lapacke are required.

OK, is the correct place to handle this dependency resolution in the new cmake build system (which announces whether a TPL is available) and the ETI specialization layer (which determines if a full specialization is available)?

> Your changes will get rid of cblas/lapacke dependency, but we are not there yet. Correct ?

I think that's right assuming I am starting to understand the ETI system. I plan on addressing this dependency in a separate PR -- I will update the description in #589 to see if we are on the same page.

These questions were really helpful, thank you.